XenModel/Actions/Updates/DownloadAndUpdateClientAction.cs

```diff
@@ -240,16 +237,36 @@ private void ValidateMsi()
 
             // Check if calculatedChecksum matches what is in chcupdates.xml
             if (!_checksum.Equals(calculatedChecksum, StringComparison.InvariantCultureIgnoreCase))
-                throw new Exception(Messages.UPDATE_CLIENT_INVALID_CHECKSUM );
+                throw new Exception(Messages.UPDATE_CLIENT_INVALID_CHECKSUM);
+
+            bool valid = false;
 
-            bool valid;
             try
             {
-                // Check digital signature of .msi
-                using (var basicSigner = X509Certificate.CreateFromSignedFile(_outputPathAndFileName))
+                log.InfoFormat("Validating certificate chain");
+                
+                var cert = new X509Certificate2(File.ReadAllBytes(_outputPathAndFileName)); // Read certificate from msi

```

Comment:
The new code does not dispose of the certificate after use as the previous code.
Also, File.ReadAllBytes is loading all the file (40MB), is there a better/alternative way to do it?